lighthouse-core/audits/dobetterweb/num-dom-nodes.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Audits a page to see how many DOM nodes it creates.
+ */
+
+'use strict';
+
+const Audit = require('../audit');
+
+class NumDOMNodes extends Audit {
+  static get MAX_DOM_NODES() {
+    return 1500;
+  }
+
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'Performance',
+      name: 'num-dom-nodes',
+      description: 'Uses a small number of DOM nodes',
+      helpText: 'Browser engineers recommend pages contain fewer than ' +
+          `~${NumDOMNodes.MAX_DOM_NODES} DOM nodes. The sweet spot is around 60 elements wide x ` +
+          '32 elements deep. A large DOM can increase memory, cause longer ' +
```

Comment:
should we be looking at length of the longest paths, or distribution of path lengths in the DOM?